components/calendar/src/any_calendar.rs

```diff
@@ -453,6 +462,68 @@ impl AnyCalendar {
         })
     }
 
+    /// Constructs an AnyCalendar for a given locale and [`AnyProvider`] data source.
+    ///
+    /// In case the locale's calendar is unknown or unspecified, it will attempt to load the default
+    /// calendar for the locale, falling back to gregorian.
+    ///
+    /// For calendars that need data, will attempt to load the appropriate data from the source.
+    ///
+    /// This API needs the `calendar/japanese@1` or `calendar/japanext@1` data key if working with Japanese calendars.
+    pub fn try_new_for_locale_with_any_provider<P>(
```

Comment:
Nit (optional): use the macro to generate these APIs. The downside is that you can't customize the docs, but the upside is that then the docs are consistent across ICU4X